src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c

```diff
@@ -23,26 +21,133 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
+    func_80B40308(_this, globalCtx);
+
+    if (_this->flt164 <= 0.0f) {
+        ActorCutscene_Stop(_this->initialActorCutscene);
+        _this->flt164 = 0.0f;
+        func_80B40080(_this);
+    }
+
+    // SFX ID
+    func_8019F1C0(&D_80B40780, (u16)0x205EU);
+}
+
+void func_80B401F8(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    Player* player;
+    Vec3f playerWorldDistance;
+
+    if (Actor_HasParent(&_this->polyActor.actor, globalCtx) != 0) {
+        _this->polyActor.actor.parent = NULL;
+        return;
+    }
+
+    player = PLAYER;
+
+    Math_Vec3f_DistXYZAndStoreDiff(&_this->collisionHeaderMinVec3f, &player->actor.world.pos, &playerWorldDistance);
+
+    if (playerWorldDistance.x >= 0.0f && playerWorldDistance.x <= _this->collisionHeaderMaxX &&
+        playerWorldDistance.z >= 0.0f && playerWorldDistance.z <= _this->collisionHeaderMaxY &&
+        fabsf(playerWorldDistance.y) < 100.0f && player->actor.yDistToWater > 12.0f) {
+        func_800B8A1C(&_this->polyActor.actor, globalCtx, 0xBA, _this->polyActor.actor.xzDistToPlayer,
+                      fabsf(_this->polyActor.actor.yDistToPlayer));
+    }
+}
+
+void func_80B40308(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    CollisionHeader* colHeader;
+    f32 sp28;
+
+    if (func_800CA1AC(globalCtx, &globalCtx->colCtx, _this->polyActor.actor.world.pos.x,
+                      _this->polyActor.actor.world.pos.z, &sp28, &colHeader) != 0) {
+        Math_Vec3s_ToVec3f(&_this->collisionHeaderMinVec3f, &colHeader->minBounds);
+        _this->collisionHeaderMaxX = colHeader->maxBounds.x;
+        _this->collisionHeaderMaxY = colHeader->maxBounds.y;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B400C8.asm")
+#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40394.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40100.asm")
+void BgGoronOyu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgGoronOyu* _this = (BgGoronOyu*)thisx;
+    void* unkPad;
+    CollisionHeader* actorCollisionHeader = NULL;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40160.asm")
+    Actor_SetScale(&_this->polyActor.actor, 0.1f);
+    BcCheck3_BgActorInit(&_this->polyActor, 1);
+    BgCheck_RelocateMeshHeader(&D_6000988, &actorCollisionHeader);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B401F8.asm")
+    _this->polyActor.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &_this->polyActor, actorCollisionHeader);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40308.asm")
+    func_80B40308(_this, globalCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40394.asm")
+    if (!!thisx->params) {
+        thisx->world.pos.y = thisx->home.pos.y;
+        func_80B40080(_this);
+        return;
+    }
```

Comment:
```suggestion
    } else {
```
You can use `if-else` to get rid of the early return